src/base/static/components/templates/map.tsx

```diff
@@ -84,11 +79,81 @@ const SpotlightMask = styled("div")({
   zIndex: 8,
 });
 
-class MapTemplate extends Component {
-  mapContainerRef = createRef();
-  addPlaceButtonRef = createRef();
+const statePropTypes = {
+  datasetsConfig: datasetsConfigPropType.isRequired,
+  hasAddPlacePermission: PropTypes.bool.isRequired,
+  hasGroupAbilitiesInDatasets: PropTypes.func.isRequired,
+  initialMapViewport: mapViewportPropType.isRequired,
+  isAddPlaceButtonVisible: PropTypes.bool.isRequired,
+  isContentPanelVisible: PropTypes.bool.isRequired,
+  isGeocodeAddressBarEnabled: PropTypes.bool.isRequired,
+  isLeftSidebarExpanded: PropTypes.bool.isRequired,
+  isRightSidebarEnabled: PropTypes.bool.isRequired,
+  isRightSidebarVisible: PropTypes.bool.isRequired,
+  isSpotlightMaskVisible: PropTypes.bool.isRequired,
+  mapSourceNames: PropTypes.arrayOf(PropTypes.string).isRequired,
```

Comment:
As mentioned below, I believe the source name is always required. If so, we can make this change:

```suggestion
  mapSourceNames: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
```